ui/js/directives/treeherder/clonejobs.js

```diff
@@ -360,17 +371,46 @@ treeherder.directive('thCloneJobs', [
                     countInfo.value = countInfo.count;
                     countInfo.title = countInfo.count + " " + countInfo.countText + " jobs in group";
                     countInfo.btnClass = countInfo.btnClass + "-count";
-                    var jobCountBtn = $(jobGroupCountInterpolator(countInfo));
-                    showHideElement(jobCountBtn, true);
+                    countInfo.visibleClass = "filter-shown";
+                    var jobCountBtn = jobGroupCountInterpolator(countInfo);
                     jobCountBtnArray.push(jobCountBtn);
                     jobCountBtnArray.push(' ');
                 }
             });
+            var output = {
+                jobBtnArray: jobBtnArray,
+                jobCountBtnArray: jobCountBtnArray
+            };
+            return output;
+        };
 
+        var addGroupJobsAndCounts = function(jgObj, platformGroup) {
+            var btnArrays = renderGroupJobsAndCounts(jgObj);
+            var jobBtnArray = btnArrays.jobBtnArray;
+            var jobCountBtnArray = btnArrays.jobCountBtnArray;
             platformGroup.find(".group-job-list").html(jobBtnArray);
             platformGroup.find(".group-count-list").html(jobCountBtnArray);
         };
 
+        var getGroupJobsAndCounts = function(jgObj) {
+            var btnArrays = renderGroupJobsAndCounts(jgObj);
+            var jobBtnArray = btnArrays.jobBtnArray;
+            var jobCountBtnArray = btnArrays.jobCountBtnArray;
+            var jobBtnHTML = "";
```

Comment:
Just a general point, I don't think it's a good idea to have so many temporary variables here. Unless the expression you're evaluating is really long or is used more than once or twice, it's better just to use it directly. i.e. you could have just done the forEach against btnArrays.jobCountBtnArray directly.